blocks/autocompleters/index.js

```diff
@@ -115,6 +115,14 @@ export function userAutocompleter() {
 					keywords: [ user.slug, user.name ],
 				};
 			} );
+			resolve( users );
+		} );
+	}, 250 );
+
+	const getOptions = ( search = false ) => {
+		const searchData = search ? { data: { search } } : null;
+		return new Promise( ( resolve ) => {
+			performSearch( resolve, searchData );
```

Comment:
Returning a promise that may never resolve makes me uncomfortable as I am unsure if it will cause memory to leak (it probably won't but it doesn't hurt to avoid the situation). I suggest that you could cause the promise to reject with a specific error (just use a string) that you can check for in `loadOptions`.

Reply:
yea, i have to agree - I feel the same way. unfortunately, this was the only way I could get the debouncing to work and part of why i had moved the debouncing up a level. debouncing a function that returns a promise is a bit problematic. 

ideally, we should debounce the function that calls the promise returning function - that way no extra promises are created. let me revisit the idea of moving this "up" again is a way that doesn't affect other autocompleters.

Reply:
@EphoxJames I tried a different approach in https://github.com/WordPress/gutenberg/pull/4406/commits/eee4ce31ee32b396ef4a788e60ac0c0cca515a9b

I've added an attribute `isDebouncedCompleter: true` to the object returned by `userAutocompleter. The in the autocomplete component, when this is set for an autocompleter, I call `debouncedLoadOptions` to debounce the async callbacks. This lets any autocompleter opt in to debouncing.

Let me know what you think of this approach.